mantidproject/mantid
# 36593 FindpeaksConvolve at Engdiff fit browser

### Description of work
Currently the find peaks algorithm can be selected through the fit browser menu Setup > Peak Finding Algorithms > FindPeaksConvovle - which will open up the algorithm GUI. A new button was added to the Eng diff fitting tab to call FindPeaksConvovle algorithm with good default values(EstimatedPeakExtent=200 and IOverSigmaThreshold=4) for Engin-X.


Fixes #36593. 


### To test:
1. Follow the testing instructions for the fitting tab of Engineering Diffraction GUI. https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html#id8
2. Click on  `Load ` button to load some data having `Add to Plot` being clicked
4. In the Plot window below, `FindPeaksConvolve` should be available with greyed out.

![image](https://github.com/mantidproject/mantid/assets/142818102/cf8f8da3-5ba4-4a88-9748-67af802eeb28)

6. Click `Fit` button to open the fit property browser and this should enable `FindPeaksConvolve` button
7. The fit property browser should have no peaks added and look similar to this

![image](https://github.com/mantidproject/mantid/assets/142818102/7bdf4356-777f-4482-bf57-d8e56fbe727a)

8. Click on `FindPeaksConvolve` to run the algorithm on the selected `Workspace` from the fit property browser with the default values as mentioned above.
10. The progress bar below the fit property browser should be updated and marked as complete at the end by adding several peak functions to the fit property browser.
![image](https://github.com/mantidproject/mantid/assets/142818102/a955f0c1-5c45-4a2f-a737-3e0200c75eae)

12. Change the `Workspace` in the property browser and click on `FindPeaksConvolve` to run the algorithm on another workspace.
13. By right-clicking on the plot window change the selected peak type to some other function type.
14. Click on the `FindPeaksConvolve` button again to replace the old peak functions with newly selected peak functions.
15. Repeat steps 10 and 11 as many times to replace the existing peak types with new types.
16. Fit->fit should initiate the fit on the added peak functions (which has been there before)
17. Click on `Fit` button on the toolbar to hide the fit property browser which should disable the `FindPeaksConvolve` button.
18. Enable `Fit` button again to show the fit property browser with the last added peaks.
19. When enabled, unticking all the `Plot` ticks in the Run selection table should also disable the `FindPeaksConvolve` button.
20. Click on `Clear` in the workspace tree should also disable the `FindPeaksConvolve` button.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
